internal/pkg/manifest/svc.go

```diff
@@ -55,6 +55,64 @@ func (r *Range) Parse() (min int, max int, err error) {
 	return aws.IntValue(r.RangeConfig.Min), aws.IntValue(r.RangeConfig.Max), nil
 }
 
+// ExposedPorts returns all the ports that are container ports available to receive traffic.
+func ExposedPorts(workloadName string, port *uint16, targetPort *int, targetContainer *string, sidecars map[string]*SidecarConfig) ([]ExposedPort, error) {
```

Comment:
Long parameter lists is also a code smell. I'll quote from the [Refactoring book from Martin Fowler](https://www.amazon.com/Refactoring-1st-first-Text-Only/dp/B004PQQRK2):

> Long parameter lists are hard to understand, because they become inconsistent and difficult to use, and because you are forever changing them as you need more data.

The solutions to this are:
> Use _Replace Parameter with Method_ when you can get the data in one parameter by making a request of an object you already know about. This object might be a field or it might be another parameter.
> Use _Preserve Whole Object_ to take a bunch of data glened from an object and replace it with the object itself.
> If you have several data items with no logical object, use _Introduce Parameter Object_.

In our situation, your original submission was the way to go with `Replace Parameter with Method`:

```go
func (lbws *LoadBalancedWebService) ExposedPorts() ([]ExposedPort, error)
func (worker *WorkerService) ExposedPorts() ([]ExposedPort, error)
```

You can remove duplication by then having each sub-type implement its own `exportPorts` method:
```go
func (lbws *LoadBalancedWebService) ExposedPorts() ([]ExposedPort, error) {
  var exposedPorts []ExposedPort
  exposedPorts = append(exposedPorts, lbws.ImageConfig.exposedPorts()...)
  exposedPorts = append(exposedPorts, lbws.RoutingRule.exposedPorts()...)
  exposedPorts = append(exposedPorts, lbws.Sidecars.exposedPorts()...)
  return exposedPorts
}


func (ws *WorkerService) ExposedPorts() ([]ExposedPort, error) {
  var exposedPorts []ExposedPort
  exposedPorts = append(exposedPorts, lbws.ImageConfig.exposedPorts()...)
  exposedPorts = append(exposedPorts, lbws.Sidecars.exposedPorts()...)
  return exposedPorts
}
```